src/NLog.Extensions.Logging/Extensions/ConfigureExtensions.cs

```diff
@@ -104,7 +104,19 @@ public static ILoggingBuilder AddNLog(this ILoggingBuilder factory, NLogProvider
         /// <returns>ILoggingBuilder for chaining</returns>
         public static ILoggingBuilder AddNLog(this ILoggingBuilder builder, LoggingConfiguration configuration)
         {
-            AddNLogLoggerProvider(builder.Services, null, null, (serviceProvider, config, options) =>
+            return AddNLog(builder, null, configuration);
+        }
+
+        /// <summary>
+        /// Enable NLog as logging provider for Microsoft Extension Logging
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="options">NLog options</param>
+        /// <param name="configuration">New NLog config.</param>
+        /// <returns>ILoggingBuilder for chaining</returns>
+        public static ILoggingBuilder AddNLog(this ILoggingBuilder builder, NLogProviderOptions options, LoggingConfiguration configuration)
```

Comment:
Thanks!

I think `AddNLog(this ILoggingBuilder builder, LoggingConfiguration configuration, NLogProviderOptions options)` (different order)  is a bit more logical for me. What do you think?

Reply:
That was my thought too initially as I was writing it. But I looked at it as the  configuration object being a secondary thing to creating the logging provider. But it may make a lot more sense in general to have the options parameter be the last in the signature, there are definitely a lot of framework methods that usually have this ordering. I have no strong feelings either way.

Reply:
please move the options as last parameter and I will merge this, thanks!

Reply:
Updated. Options moved to be the last.

Reply:
Thanks a lot!